lib/generator/template_values.rb

```diff
@@ -3,20 +3,27 @@ module Generator
   class TemplateValues
```

Comment:
The original intention of `TemplateValues` was that it it should be a dumb container and would only contain the template values available to the erb file. The `initialize` method provided one place where it was possible to see exactly all the properties that would be available to the erb. 

It is the responsibility of the `TemplateValuesFactory` to construct the necessary values.

If that is the goal, then there is probably a nicer way of implementing this but I'm not sure this commit is heading in that direction.